File: client/src/game/engine/casualtiesCalculator.ts
// src/game/engine/casualtiesCalculator.ts

import Scenario from "@/game/Scenario";
import Aircraft from "@/game/units/Aircraft";
import Airbase from "@/game/units/Airbase";
import Facility from "@/game/units/Facility";
import Ship from "@/game/units/Ship";

// Define a type that includes all units that can be considered casualties
type CasualtyUnit = Aircraft | Ship | Facility | Airbase;

/**
 * Increments the casualty counter for a side when one of its units is lost.
 * @param scenario The current game scenario.
 * @param lostUnit The unit that was destroyed.
 */
export function incrementCasualty(scenario: Scenario, lostUnit: CasualtyUnit) {
    const side = scenario.getSide(lostUnit.sideId);
    if (side) {
        side.casualties = (side.casualties || 0) + 1;
        console.log(`${side.name} suffered a casualty. Total Casualties: ${side.casualties}`);
    }
}

File: gym/blade/engine/casualtiesCalculator.py
from typing import Union
from blade.Scenario import Scenario
from blade.units.Aircraft import Aircraft
from blade.units.Airbase import Airbase
from blade.units.Facility import Facility
from blade.units.Ship import Ship

# Define a type alias for units that can be considered casualties
CasualtyUnit = Union[Aircraft, Ship, Facility, Airbase]

def increment_casualty(scenario: Scenario, lost_unit: CasualtyUnit):
    """
    Increments the casualty counter for a side when one of its units is lost.
    
    Args:
        scenario (Scenario): The current game scenario.
        lost_unit (CasualtyUnit): The unit that was destroyed.
    """
    side = scenario.get_side(lost_unit.side_id)
    if side:
        side.casualties = getattr(side, 'casualties', 0) + 1
        print(f"{side.name} suffered a casualty. Total Casualties: {side.casualties}")
